core/src/main/java/org/apache/druid/java/util/http/client/pool/ResourcePool.java

```diff
@@ -189,6 +191,46 @@ private ImmediateCreationResourceHolder(
         );
       }
     }
+  }
+
+  private static class LazyCreationResourceHolder<K, V> extends ResourceHolderPerKey<K, V>
+  {
+    private LazyCreationResourceHolder(
+        int maxSize,
+        long unusedResourceTimeoutMillis,
+        K key,
+        ResourceFactory<K, V> factory
+    )
+    {
+      super(maxSize, unusedResourceTimeoutMillis, key, factory);
```

Comment:
Are we not initializing even a single connection here?

Reply:
Are there any potential problems that may arise from not initializing a subset of resources eagerly?